### PR TITLE
ci: skip running test-e2e on PRs from forks 

### LINKED ISSUE #15557: forks are unable to run ci because no access to secrets
### Problem Description
Forked repositories cannot run CI workflows due to restrictions on accessing a sensitive license key required for E2E tests. This limitation blocks contributions from forks and prevents full validation of pull requests.  

### Desired Solution
Implement a secure mechanism to allow forks to pass CI workflows while protecting the license key.  

### Potential Approaches  
1. **Skip E2E tests for forks**: Run non-secret-dependent jobs for forks and reserve E2E tests for workflows triggered in the base repository.  
    ```yaml
    jobs:
      e2e-tests:
        if: github.repository == 'org/repo' # Run only in the base repository
    ```
2. **Ephemeral Licenses**: Generate short-lived license keys for PRs from forks, valid only for the duration of the CI workflow.

### PR DESCRIPTION
**Pull request by matifali**
A workaround for #15557. We will come up with a more permanent solution

### CODE REVIEW

**Review by dannykopping — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -643,6 +640,9 @@ jobs:
             name: test-e2e
           - premium: true
             name: test-e2e-premium
+    # Skip test-e2e on forks as they don't have access to CI secrets
+    if: (needs.changes.outputs.go == 'true' || needs.changes.outputs.ts == 'true' || needs.changes.outputs.ci == 'true') && !(github.event.pull_request.head.repo.fork && matrix.variant.premium) || github.ref == 'refs/heads/main'
```
Comment (deansheather) [inline .github/workflows/ci.yaml:644]:
```suggestion
    if: (needs.changes.outputs.go == 'true' || needs.changes.outputs.ts == 'true' || needs.changes.outputs.ci == 'true' || github.ref == 'refs/heads/main') && !(github.event.pull_request.head.repo.fork && matrix.variant.premium) 
```

Comment (matifali) [reply .github/workflows/ci.yaml:644]: good catch. Thank you.

**Comment by matifali** [comment]
@deansheather looks like matrix context is not available at job level.
```
[Invalid workflow file: .github/workflows/ci.yaml#L644](https://github.com/coder/coder/actions/runs/12232484795/workflow)
The workflow is not valid. .github/workflows/ci.yaml (Line: 644, Col: 9): Unrecognized named-value: 'matrix'. Located at position 198 within expression: (needs.changes.outputs.go == 'true' || needs.changes.outputs.ts == 'true' || needs.changes.outputs.ci == 'true' || github.ref == 'refs/heads/main') && !(github.event.pull_request.head.repo.fork && matrix.variant.premium)
```

**Comment by deansheather** [comment]
@matifali maybe just make it skip e2e entirely for now to unblock external contributions, and we can prioritise coming up with a solution (like generating short licenses each time)

**Pull request merged**